kquery/KQuery.g4

```diff
@@ -0,0 +1,357 @@
+    | LeftParen bitExtractExpr widthOrSizeExpr expression RightParen                        #BitExtractExprWidth
+    | LeftParen genericBitRead widthOrSizeExpr expression version RightParen                #ReadExpresssionVersioned
+    | LeftParen selectExpr widthOrSizeExpr leftExpr rightExpr expression RightParen         #SelectExprWidth
+    | LeftParen exprNegation widthOrSizeExpr expression RightParen                          #NegationExprWidth
+    | LeftParen exprNegation expression RightParen                                          #NegetionExpr
+    | LeftParen genericBitRead widthOrSizeExpr expression RightParen                        #ReadExpr
+    | version                                                                               #VersionExpr
+    | number                                                                                #Singleton
+    ;
+
+genericBitRead
+    : READ          
+    | READLSB
+    | READMSB
+    ;
+
+bitExtractExpr
+    : ZEXT
+    | SEXT
+    ;
+    
+version
+    : varName                                               #VersionVariableName
+    | namedConstant Colon expression                        #VersionedNamedAbbreviation
+    | LeftBracket updateList RightBracket ATR version       #UpdationList
+    | LeftBracket RightBracket ATR version                  #NoUpdateList
+    ;
+    
+notExpr
+    : NOT
+    ;
+    
+concatExpr
+    : CONCAT
+    ;
+
+exprNegation
+    : NEGETION
+    ;
+
+selectExpr
+    : SELECT
+    ;
+
+arrExtractExpr
+    : EXTRACT
+    ;
+    
+varName
+    : Identifier
+    ;
+    
+leftExpr 
+    : expression
+    ; 
+
+rightExpr
+    : expression
+    ;
+    
+namedConstant
+    : Identifier
+    ;
+    
+updateList 
+    : expression Equal expression COMMA updateList
+    | expression Equal expression
+    ;
+
+bitwiseExpr 
+    : BITWISEAND 
+    | BITWISEOR 
+    | BITWISEXOR 
+    | SHL 
+    | LSHR 
+    | ASHR
+    ;
+    
+comparisonExpr 
+    : EQ 
+    | NEQ 
+    | ULT
+    | UGT
+    | ULE 
+    | UGE 
+    | SLT 
+    | SLE 
+    | SGT 
+    | SGE
+    ;
+    
+arithmeticExpr 
+    : ADD
+    | SUB 
+    | MUL 
+    | UDIV 
+    | UREM 
+    | SDIV 
+    | SREM
+    ;
+    
+domain : widthOrSizeExpr ;
+rangeLimit : widthOrSizeExpr ;
+arrName : Identifier ;
+
+numberList
+    : number
+    | number numberList
+    ;
+
+number 
+    : boolnum 
+    | signedConstant
+    | constant
+    ;
+
+constant: Constant;
+boolnum: Boolean;
+signedConstant: SignedConstant;
+
+Boolean
+    : TrueMatch
+    | FalseMatch
+    ;
+    
+SignedConstant 
+    : (PLUS | MINUS)Constant
+    ;
+    
+Constant
+    : DIGIT+ 
+    | BinConstant 
+    | OctConstant 
+    | HexConstant
+    ;
+    
+BinConstant 
+    : BinId (BIN_DIGIT)+  
+    ;
+    
+OctConstant 
+    : OctId (OCTAL_DIGIT)+  
+    ;
+    
+HexConstant 
+    : HexId (HEX_DIGIT)+
+    ;
+
+FloatingPointType 
+    : FP DIGIT ((.).*?)?  
```

Comment:
This rule is wrong. I looked at the original Spec and it says `floating-point-type = "fp[0-9]+([.].*)?"` 

`(.)` is not the same as `[.]`--which is exactly what you should be using for your rule.

Further, you have `fp` as a token, but the definition "shields" `Identifier`. Can `fp` be an identifier? If so, you need to define a rule `identifier: Identifier | fp | .......`. The same thing may apply to other rules, e.g., "INT: 'i';".